runtime/reconcilers/project_parser.go

```diff
@@ -291,6 +291,13 @@ func (r *ProjectParserReconciler) reconcileParser(ctx context.Context, inst *dri
 	}
 
 	// not setting restartController=true when diff is actually nil prevents infinite restarts
+	if diff == nil {
+		err := r.reconcileProjectConfig(ctx, parser, false)
+		if err != nil {
+			return err
+		}
+	}
+
 	restartController := diff != nil && (diff.ModifiedDotEnv || diff.Reloaded)
 	if restartController {
 		return r.reconcileProjectConfig(ctx, parser, true)
```

Comment:
If you don't realize the dependency between `diff == nil` and `restartController`, it looks like `reconcileProjectConfig` can be called twice in a row here. This might help make things clearer:

```suggestion
	updateConfig := diff == nil || diff.ModifiedDotEnv || diff.Reloaded
	if updateConfig {
		restartController := diff != nil
		err := r.reconcileProjectConfig(ctx, parser, restartController)
		if err != nil {
			return err
		}
		if restartController {
			return nil
		}		
```

Reply:
Yeah looks more cleaner.